test_image.sh

```diff
@@ -77,15 +79,31 @@ function prepare_mount {
 
 	cp -r templates/test/* "${TEST_DIR}"
 
-	if [ -n "${LIFERAY_DOCKER_TEST_HOTFIX_URL}" ]
+	mkdir -p "${TEST_DIR}/patching"
+
+	if [ -n "${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL}" ]
 	then
-		mkdir -p "${TEST_DIR}/patching"
+		local patcing_tool_file_name=${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL##*/}
+
+		download "downloads/patching-tool/${patcing_tool_file_name}" "${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL}"
+	else
+		local patcing_tool_file_name=$(basename $(ls -Art downloads/patching-tool/*.zip | tail -n 1))
+	fi
 
+	if [ -n "${LIFERAY_DOCKER_TEST_HOTFIX_URL}" ]
+	then
 		local hotfix_file_name=${LIFERAY_DOCKER_TEST_HOTFIX_URL##*/}
 
 		download "downloads/hotfix/${hotfix_file_name}" "${LIFERAY_DOCKER_TEST_HOTFIX_URL}"
 
 		cp "downloads/hotfix/${hotfix_file_name}" "${TEST_DIR}/patching"
+
+		if [ ! -e "downloads/patching-tool/patching-tool-${LIFERAY_DOCKER_TEST_PATCHING_TOOL_VERSION}.zip" ]
+		then
+			download "downloads/patching-tool/patching-tool-${LIFERAY_DOCKER_TEST_PATCHING_TOOL_VERSION}.zip" "files.liferay.com/private/ee/fix-packs/patching-tool/patching-tool-${LIFERAY_DOCKER_TEST_PATCHING_TOOL_VERSION}.zip"
```

Comment:
I think we should skip the test (pass by default) if the patching tool URL was not set. It would be weird to have Patching Tool failures on CE images. So I don't think this download is necessary.

Reply:
The test had a skip logic in place, if the variable is not defined. However this check was not implemented in the download process. Fixed